sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2100,7 +2100,7 @@ object SQLConf {
       .stringConf
       .createOptional
 
-  val LEGACY_LOOSE_UPCAST = buildConf("spark.sql.legacy.looseUpcast")
+  val LEGACY_LOOSE_UPCAST = buildConf("spark.sql.legacy.looseUpcast.enabled")
```

Comment:
`spark.sql.legacy.doLooseUpcast`? Then we can avoid creating the `looseUpcast` namespace.